From 1aaef52d5d11c18a62e5f83271a5b33743da7b9a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <karel.koci@nic.cz>
Date: Thu, 16 May 2019 15:52:45 +0200
Subject: [PATCH] base-files: run switch-brach --verify after update

When base-files is updated it replaces its distfeeds to default. That
moves it from selected branch back to deploy. We are unable to do that
change only on server because some branches are just binary copies of
less stable branch. This solves it just by setting current branch back
in distfeeds.

We use --verify to not nest updater runs and --force to set any branch
user chooses not only standard ones.
---
 package/base-files/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/package/base-files/Makefile b/package/base-files/Makefile
index 3e24f50c4e..5677202e13 100644
--- a/package/base-files/Makefile
+++ b/package/base-files/Makefile
@@ -41,7 +41,7 @@ define Package/base-files
   CATEGORY:=Base system
   DEPENDS:= \
 	+netifd +libc +jsonfilter +SIGNED_PACKAGES:usign +SIGNED_PACKAGES:openwrt-keyring \
-	+NAND_SUPPORT:ubi-utils +fstools +fwtool \
+	+NAND_SUPPORT:ubi-utils +fstools +fwtool +switch-branch \
 	+SELINUX:procd-selinux +!SELINUX:procd +USE_SECCOMP:procd-seccomp \
 	+SELINUX:busybox-selinux +!SELINUX:busybox
   TITLE:=Base filesystem for OpenWrt
@@ -275,6 +275,8 @@ define Package/base-files/postinst
 			"/etc/init.d/$$L" enable
 		fi
 	done < /etc/services_wanted
+	# Update new /etc/opkg/distfeeds.conf to point to correct repository
+	switch-branch --force --verify
 }
 endef
 
-- 
2.47.1

